executor/write.go

```diff
@@ -147,6 +159,22 @@ func updateRecord(ctx context.Context, h int64, oldData, newData []types.Datum,
 	return true, nil
 }
 
+func checkHandleExists(ctx context.Context, t table.Table, recordID int64) error {
+	txn := ctx.Txn()
+	// Check key exists.
+	recordKey := t.RecordKey(recordID)
+	e := kv.ErrKeyExists.FastGen("Duplicate entry '%d' for key 'PRIMARY'", recordID)
+	txn.SetOption(kv.PresumeKeyNotExistsError, e)
+	defer txn.DelOption(kv.PresumeKeyNotExistsError)
```

Comment:
Above two lines can be removed.
It's only used for INSERT statement when `kv.PresumeKeyNotExists` option is set.